src/Sulu/Bundle/AdminBundle/Resources/js/containers/Form/stores/ResourceFormStore.js

```diff
@@ -87,15 +88,145 @@ export default class ResourceFormStore extends AbstractFormStore implements Form
         });
     };
 
-    @action handleSchemaResponse = ([schema, jsonSchema]: [Schema, Object]) => {
+    handleSchemaResponse = ([schema, jsonSchema]: [Schema, Object]) => {
         this.validator = jsonSchema ? ajv.compile(jsonSchema) : undefined;
         this.pathsByTag = {};
 
-        this.schema = schema;
-        this.addMissingSchemaProperties();
-        this.setSchemaLoading(false);
+        this.loadAndMergeOriginData(this.schema, schema).then(action(()=> {
+            this.removeObsoleteData(this.schema, schema, this.data);
+            this.schema = schema;
+            this.addMissingSchemaProperties();
+            this.setSchemaLoading(false);
+        }));
     };
 
+    loadAndMergeOriginData(oldSchema: Schema, newSchema: Schema) {
+        if (oldSchema && toJS(this.type) === this.data['originTemplate']){
```

Comment:
can we add a comment that explains the `toJS(this.type) === this.data['originTemplate']` condition?